### PR TITLE
e2e tests with cypress

### LINKED ISSUE #164: E2E smoke tests to avoid regressions on popular browsers
As a Selenium runtime I recommend https://www.browserstack.com/open-source

Browsers: Latest Chrome, Firefox, Safari and Edge
Systems: Windows 10 and Latest Mac
Scenarios: main page and report pages for every visualisation type (using wikipedia example)
Assertions: no JS errors, the most important components rendered

### PR DESCRIPTION
**Pull request by adrianmroz**
Closes #164

### CODE REVIEW

File: cypress/integration/home-page.spec.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+context('Home Page', () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:9090');
+  });
+
+  it('should load wikipedia cube', () => {
+    cy.get('.cube-header-bar .title')
+      .should('contain', 'Wikipedia Example');
+  });
```
Comment (bigsergey) [inline cypress/integration/home-page.spec.js]: Is `cy.get` promise? 

Does Cypress require to return `cy` when async/await notation is not used?
```js
it('should load wikipedia cube', () =>  cy
  .get('.cube-header-bar .title')
  .should('contain', 'Wikipedia Example'));
```

File: package.json [hunk 2]
```diff
@@ -54,7 +54,12 @@
     "tsc": "tsc",
     "lint": "npm-run-all -p lint:*",
     "lint:ts": "tslint -p . --format verbose",
-    "lint:sass": "sass-lint -v"
+    "lint:sass": "sass-lint -v",
+    "start:examples": "npm run start -- --examples",
+    "e2e": "start-server-and-test start:examples http://localhost:9090/health/ready cy:run",
+    "e2e:dev": "start-server-and-test start:examples http://localhost:9090/health/ready cy:open",
```
Comment (bigsergey) [inline package.json]: `e2e:watch`?

File: package.json [hunk 3]
```diff
@@ -54,7 +54,12 @@
...
+    "cy:run": "cypress run",
+    "cy:open": "cypress open"
```
Comment (bigsergey) [inline package.json]: Do you really need this scripts?

File: package.json (hunk 2 above)
Comment (adrianmroz) [reply package.json]: It runs different mode of cypress and I like "dev" suffix.

File: package.json (hunk 3 above)
Comment (adrianmroz) [reply package.json]: I thought so, but I could just quote commands in `start-server-and-test`!

File: cypress/integration/home-page.spec.js (hunk 1 above)
Comment (adrianmroz) [reply cypress/integration/home-page.spec.js]: It is something more complicated. It runs multiple times so more like a stream. Anyways, I shouldn't return them - I will get warning in console. And awaiting on them is also not good practice (but possible) so I rewrote last test.

Comment (bigsergey) [reply cypress/integration/home-page.spec.js]: 👍

File: cypress.json [hunk 4]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (bigsergey) [inline cypress.json]: Do you need it?

Comment (adrianmroz) [reply cypress.json]: `cypress` adds it on running, I don't want to gitignore it because it is supposed to be used for overrides ...

**Pull request merged**